src/libsync/configfile.cpp

```diff
@@ -846,9 +846,9 @@ void ConfigFile::setClientVersionString(const QString &version)
     settings.setValue(clientVersionC(), version);
 }
 
-std::unique_ptr<QSettings> ConfigFile::settingsWithGroup(const QString &group, QObject *parent)
+std::unique_ptr<QSettings> ConfigFile::settingsWithGroup(const QString &group)
 {
-    std::unique_ptr<QSettings> settings(new QSettings(ConfigFile::configFile(), QSettings::IniFormat, parent));
```

Comment:
Why did you remove the `parent` argument?

Reply:
I just wanted to ask the same thing

Reply:
Whoops, that parent might result in a double delete when the parent gets destroyed before the unique ptr's constructor is run.

Reply:
hmm right.

I hope we properly use the function and don't keep a pointer in some member...

Reply:
I'll check that later this morning.

Reply:
We're fine, and it's good to get rid of the parent argument.

Reply:
> I just wanted to ask the same thing

There was a mixed approach of object ownership: both parent and unique_ptr - this shouldn't be the case, they aren't interoperable. Double-delete problem will occur.